### ISSUE TITLE
Generalize one-argument ak.where to variable length arrays

### ISSUE THREAD

**Issue by deinal** [scikit-hep/awkward#801]
ak.where works with one argument for regularly shaped arrays,
```
B = ak.Array([[False, True], [True, False]])
ak.where(B)
>>> (<Array [0, 1] type='2 * int64'>, <Array [1, 0] type='2 * int64'>)
```
but not for variable length ones:
```
A = ak.Array([[False, True], [True, False, True]])
ak.where(A)
>>> ValueError: in ListOffsetArray64, cannot convert to RegularArray because subarray lengths are not regular
```
A crude workaround is using ak.where with multiple arguments:
```
C = ak.where(A, ak.local_index(A), ak.broadcast_arrays(A, -1)[1])
C[C != -1]
>>> <Array [[1], [0, 2]] type='2 * var * int64'>
```
The output of `ak.where(A)` should probably also be the same as for the workaround, and not a tuple of arrays as we saw above for `ak.where(B)`.

**Comment by jpivarski** [comment]
The only problem that I see is that the output type is so different from what NumPy returns. But then again, NumPy's output can only ever make sense for regular arrays: the tuple of `i` coordinates and `j` coordinates is only useful if the two line up, which they wouldn't fit an irregular array.

Since the output is so different, maybe this new feature should be a new function with a new name? NumPy has already taken both one-argument `where` and `nonzero` for this same purpose (with the same output type: a tuple of aligned arrays), but maybe something like `argtrue`?

**Comment by deinal** [comment]
Yes you're right, the non-tuple version should have another name such as `argtrue`.

How about `ak.where` generalizing for variable length arrays, as long as the depth remains the same for every subarray?
```python
A = ak.Array([[False, True], [True, False, True]])
ak.where(A)
>>> (<Array [0, 1, 1] type='3 * int64'>, <Array [1, 0, 2] type='3 * int64'>)
```
We then get the expected:
```python
A[ak.where(A)]
>>> <Array [True, True, True] type='3 * bool'>
```
Another example with depth three:
```python
ak.where(ak.Array([[[False, True], [True]], [[False, False], [True, False, True]]]))
>>> (<Array [0, 0, 1, 1] type='4 * int64'>, 
...  <Array [0, 1, 1, 1] type='4 * int64'>, 
...  <Array [1, 0, 0, 2] type='4 * int64'>)
```

**Comment by jpivarski** [comment]
That would work, though it would take some thinking to figure out how to implement this.

It could also be `ak.nonzero`, as that should have the same behavior.

**Comment by deinal** [comment]
Absolutely!  `ak.where` with one argument could just call that then
